src/longRunningCalls/longrunning.ts

```diff
@@ -161,37 +193,66 @@ export class Operation extends EventEmitter {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const self = this;
     const operationsClient = this.longrunningDescriptor.operationsClient;
-
     function promisifyResponse() {
       if (!callback) {
         return new Promise((resolve, reject) => {
-          if (self.latestResponse.error) {
-            const error = new GoogleError(self.latestResponse.error.message!);
-            error.code = self.latestResponse.error.code!;
-            reject(error);
+          if ('done' in self.latestResponse) {
```

Comment:
If I understand correctly, `if ('done' in self.latestResponse)` means "if not diregapic", right?  Maybe we could have a helper function to make the logic more readable?

Reply:
Also, an alternative way of doing that is 

```
if (this.diregapic) {
  const response = self.latestResponse as ComputeSomething; // force the type
  ...
} else {
  const response = self.latestResponse as WhateverNonComputeTypeItIs;
  ...
}
```

Not sure what would be more readable.